samples/form_validate/src/sample/form.cljd

```diff
@@ -0,0 +1,30 @@
+(ns sample.form
+  "Faithful port of https://docs.flutter.dev/cookbook/forms/validation"
+  (:require ["package:flutter/material.dart" :as m]
+            [cljd.flutter :as f]
+            [cljd.string :as str]))
+
+(defn my-custom-form []
+  (let [form-key (#/(m/GlobalKey m/FormState))]
+    (f/widget
+     :get [m/ScaffoldMessenger]
+     (m/Form .key form-key)
+     (m/Column .crossAxisAlignment m/CrossAxisAlignment.start)
+     .children
+     [(m/TextFormField
+       .validator (fn [value] (when (str/blank? value) "Please enter some text")))
+      (m/Padding .padding (m/EdgeInsets.symmetric .vertical 16.0))
+      (m/ElevatedButton
+       .onPressed #(when (.validate ^m/FormState (.-currentState form-key))
```

Comment:
you don't need the type hint here, we are able to infer it

Reply:
Somehow along the way as I was making changes, I got a compiler warning (and it actually took me some thinking to figure out what the hint even needed to be), but you are right, it now works without.